test/e2e/e2e_test.go

```diff
@@ -55,15 +59,35 @@ var _ = ginkgo.Describe("TrainJob e2e", func() {
 			// Wait for TrainJob to be in Succeeded status.
 			ginkgo.By("Wait for TrainJob to be in Succeeded status", func() {
 				gomega.Eventually(func(g gomega.Gomega) {
-					gotTrainJob := &trainer.TrainJob{}
-					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(trainJob), gotTrainJob)).Should(gomega.Succeed())
-					g.Expect(gotTrainJob.Status.Conditions).Should(gomega.BeComparableTo([]metav1.Condition{
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(trainJob), trainJob)).Should(gomega.Succeed())
+					g.Expect(trainJob.Status.Conditions).Should(gomega.BeComparableTo([]metav1.Condition{
 						{
-							Type:    trainer.TrainJobCreated,
+							Type:    trainer.TrainJobComplete,
 							Status:  metav1.ConditionTrue,
-							Reason:  trainer.TrainJobJobsCreationSucceededReason,
-							Message: constants.TrainJobJobsCreationSucceededMessage,
+							Reason:  jobsetconsts.AllJobsCompletedReason,
+							Message: jobsetconsts.AllJobsCompletedMessage,
 						},
+					}, util.IgnoreConditions))
+				}, util.TimeoutE2E, util.Interval).Should(gomega.Succeed())
+			})
+		})
+		// TODO (andreyvelich): This should be changed to DeepSpeed and MLX runtimes.
+		// Verify `mpi-distributed` ClusterTrainingRuntime.
+		ginkgo.It("should create TrainJob with MPI runtime reference", func() {
```

Comment:
I do not think that this works well since as I described in https://github.com/kubeflow/trainer/pull/2481, the current MPI plugin does not work well and does not align with OpenMPI specification.

To rapidly support depensOn, I would recommend removing this case.

Reply:
In that case, shouldn't torch be enough?

Reply:
Torch doesn't test `DependsOn` API. If you want I can remove this e2e, and we will add it once you add support for MPI.

Reply:
Is there any blocker to verify dependsOn in torch TrainJob e2e?

> f you want I can remove this e2e, and we will add it once you add support for MPI.

Yes.

Reply:
>Is there any blocker to verify dependsOn in torch TrainJob e2e?

Let's post-pone it to the next PRs.

Reply:
I'm ok with next PR. Basically, JobSet is responsible for dependsOn verification instead of our side.
We can focus only on combination tests (TrainJob + Runtime + DependsOn)